core/src/main/scala/org/apache/spark/mapred/SparkHadoopMapRedUtil.scala

```diff
@@ -70,7 +70,8 @@ object SparkHadoopMapRedUtil extends Logging {
       if (shouldCoordinateWithDriver) {
         val outputCommitCoordinator = SparkEnv.get.outputCommitCoordinator
         val taskAttemptNumber = TaskContext.get().attemptNumber()
-        val canCommit = outputCommitCoordinator.canCommit(jobId, splitId, taskAttemptNumber)
+        val stageId = TaskContext.get().stageId()
+        val canCommit = outputCommitCoordinator.canCommit(stageId, splitId, taskAttemptNumber)
```

Comment:
It make sense to use stageId there, since before jobId was used instead of stageId. I will test that.

Reply:
Removing jobId from the signature of commitTask will cause a binary incompatibility error, since commitTask here is a public method. So we will ended up with a parameter that will stay unused.

Reply:
Ah, great, another public class that has no good reason to be public...

Reply:
Ever thought of returning why the commit was refused, e.g: unknown stage ID vs other task attempt committed, vs you ar considered failed?
Not that the task committer should behave differently, but it might be nice to pass that info back for logging alone

Reply:
Thanks @steveloughran for the coment. The reason will be logged in method handleAskPermissionToCommit in org.apache.spark.scheduler.OutputCommitCoordinator.scala